poetry/repositories/pypi_repository.py

```diff
@@ -432,14 +432,17 @@ def _get_info_from_urls(self, urls: Dict[str, List[str]]) -> "PackageInfo":
     def _get_info_from_wheel(self, url: str) -> "PackageInfo":
         from poetry.inspection.info import PackageInfo
 
+        pathsplits = urllib.parse.urlparse(url).path.rsplit("/")
```

Comment:
I think this logic is messy and could be greatly improved with pathlib.

e.g.

```python
pathlib.PurePath(urllib.parse.urlparse("https://my.pypi.server/path/to/my_package.whl/").path).name # 'my_pacakge.whl'
```

Also, let's factor it out into a `@staticmethod def _helper` and re-use it to DRY this up.